packages/react-reconciler/src/__tests__/ReactHooksWithNoopRenderer-test.internal.js

```diff
@@ -491,6 +491,50 @@ describe('ReactHooksWithNoopRenderer', () => {
       ]);
       expect(ReactNoop.getChildren()).toEqual([span(22)]);
     });
+
+    it('warns about render phase update on a different component', async () => {
```

Comment:
I presume there's already a test for it _not_ warning if it's the same component?